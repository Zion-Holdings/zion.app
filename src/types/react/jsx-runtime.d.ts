declare module 'react/jsx-runtime' {
  export function jsx(...args: any[]): any;
  export const jsxs: any;
  export const Fragment: any;
}
