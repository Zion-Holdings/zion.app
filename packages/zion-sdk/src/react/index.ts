export { ZionProvider, useZion } from './provider';
export { useJobs, useWalletBalances } from './hooks';