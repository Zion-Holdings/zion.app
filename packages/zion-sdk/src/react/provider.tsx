import React, { createContext, useContext, useMemo } from 'react';
import { ZionClient, ZionClientOptions } from '../zion-client';

const ZionContext = createContext<ZionClient | null>(null);

export function ZionProvider({ options, children }: { options: ZionClientOptions; children: React.ReactNode }) {
  const client = useMemo(() => new ZionClient(options), [options.baseUrl, options.apiKey]);
  return <ZionContext.Provider value={client}>{children}</ZionContext.Provider>;
}

export function useZion() {
  const ctx = useContext(ZionContext);
  if (!ctx) throw new Error('useZion must be used within a ZionProvider');
  return ctx;
}