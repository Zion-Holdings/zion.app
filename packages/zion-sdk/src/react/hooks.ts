import { useEffect, useState } from 'react';
import { useZion } from './provider';
import type { Job, WalletBalance } from '../types';

export function useJobs() {
  const zion = useZion();
  const [data, setData] = useState<Job[] | null>(null);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState<Error | null>(null);
  useEffect(() => {
    let mounted = true;
    setLoading(true);
    zion.marketplace
      .listJobs({ limit: 10 })
      .then((res) => mounted && setData(res))
      .catch((e) => mounted && setError(e as Error))
      .finally(() => mounted && setLoading(false));
    return () => {
      mounted = false;
    };
  }, [zion]);
  return { data, loading, error };
}

export function useWalletBalances(address?: string) {
  const zion = useZion();
  const [data, setData] = useState<WalletBalance[] | null>(null);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState<Error | null>(null);
  useEffect(() => {
    let mounted = true;
    setLoading(true);
    zion.token
      .getBalances(address)
      .then((res) => mounted && setData(res))
      .catch((e) => mounted && setError(e as Error))
      .finally(() => mounted && setLoading(false));
    return () => {
      mounted = false;
    };
  }, [zion, address]);
  return { data, loading, error };
}