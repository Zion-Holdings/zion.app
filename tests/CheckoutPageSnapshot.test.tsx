import { render, screen } from '@testing-library/react';'import { MemoryRouter } from 'react-router-dom';'import Checkout from '@/src/pages/Checkout';'import React from 'react';''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
jest.mock('react-redux', () => ({'  ...jest.requireActual('react-redux'),'  useSelector: jest.fn().mockImplementation((sel) => sel({ cart: { items: [] } })),'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
}));

test('Checkout page renders header', () => {'  const { _asFragment } = render('''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
    <MemoryRouter>
      <Checkout />
    </MemoryRouter>
  );

  expect(screen.getByRole('heading', { name: /checkout/i })).toBeInTheDocument();'  expect(asFragment()).toMatchSnapshot();'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
});
