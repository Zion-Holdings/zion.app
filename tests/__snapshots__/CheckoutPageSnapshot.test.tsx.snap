// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkout page renders header 1`] = `
<DocumentFragment>
  <div
    class="container max-w-2xl py-10"
  >
    <h1
      class="text-3xl font-bold mb-6"
    >
      Checkout
    </h1>
    <div
      class="mb-4"
    >
      <h2
        class="font-semibold mb-2"
      >
        Order Summary
      </h2>
      <ul
        class="space-y-1"
      />
      <div
        class="flex justify-between font-semibold mt-2"
      >
        <span>
          Subtotal
        </span>
        <span>
          $0.00
        </span>
      </div>
    </div>
    <form
      class="space-y-4"
    >
      <div
        class="space-y-2"
      >
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for=":r0:-form-item"
        >
          Name
        </label>
        <input
          aria-describedby=":r0:-form-item-description"
          aria-invalid="false"
          class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm text-gray-900 dark:text-gray-100 file:text-gray-900 dark:file:text-gray-100"
          id=":r0:-form-item"
          name="name"
          value=""
        />
      </div>
      <div
        class="space-y-2"
      >
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for=":r1:-form-item"
        >
          Email
        </label>
        <input
          aria-describedby=":r1:-form-item-description"
          aria-invalid="false"
          class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm text-gray-900 dark:text-gray-100 file:text-gray-900 dark:file:text-gray-100"
          id=":r1:-form-item"
          name="email"
          type="email"
          value=""
        />
      </div>
      <div
        class="space-y-2"
      >
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for=":r2:-form-item"
        >
          Address
        </label>
        <input
          aria-describedby=":r2:-form-item-description"
          aria-invalid="false"
          class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm text-gray-900 dark:text-gray-100 file:text-gray-900 dark:file:text-gray-100"
          id=":r2:-form-item"
          name="address"
          value=""
        />
      </div>
      <div
        class="space-y-2"
      >
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for=":r3:-form-item"
        >
          City
        </label>
        <input
          aria-describedby=":r3:-form-item-description"
          aria-invalid="false"
          class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm text-gray-900 dark:text-gray-100 file:text-gray-900 dark:file:text-gray-100"
          id=":r3:-form-item"
          name="city"
          value=""
        />
      </div>
      <div
        class="space-y-2"
      >
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for=":r4:-form-item"
        >
          Country
        </label>
        <input
          aria-describedby=":r4:-form-item-description"
          aria-invalid="false"
          class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm text-gray-900 dark:text-gray-100 file:text-gray-900 dark:file:text-gray-100"
          id=":r4:-form-item"
          name="country"
          value=""
        />
      </div>
      <div />
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-transform hover:scale-105 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 w-full"
        disabled=""
        type="submit"
      >
        Pay $0.00
      </button>
    </form>
    <p
      class="text-xs text-zion-slate-light"
    >
      Use test card 4242 4242 4242 4242 with any future date and CVC.
    </p>
  </div>
</DocumentFragment>
`;
